Automattic/wp-calypso
# Replace Jetpack Search discount with Sales Coupon

#### Changes proposed in this Pull Request

* Use product sales coupon to calculate Jetpack Search discount
* Remove `useCouponDiscount` as it is no longer needed.

#### Testing instructions

1. Boot branch on Calypso Green.
2. Navigate to `/pricing`.
3. Verify that production matches the branch for the Jetpack Search discount.
4. Sandbox `public-api.wordpress.com` and apply D77000-code
5. Verify that the Jetpack sale is not applied twice to any product on the pricing page
   - Most Yearly Jetpack products & plans should be an additional 20% off, Jetpack Search should be 60% off

<img width="1030" alt="Screen Shot 2022-03-17 at 11 30 29" src="https://user-images.githubusercontent.com/2810519/158871706-22c95165-a645-41b7-84e9-289ff147d77f.png">
